app/ts/components/IPFSSubpathRedirect.tsx

```diff
@@ -0,0 +1,36 @@
+import { CID } from 'multiformats'
+import { useEffect } from 'preact/hooks'
+
+export const IPFSSubpathRedirect = () => {
+	function extractHashFromBasePath() {
+		const htmlBase = document.querySelector('base')
+		if (!htmlBase || !htmlBase.href.includes('/ipfs/')) return
+		const maybeCidMatch = /\/ipfs\/(.*?)\//.exec(htmlBase.href)
+		return maybeCidMatch?.at(1)
+	}
+
+	function generateIPFSRedirectUrl() {
+		const cidFromPath = extractHashFromBasePath()
+
+		if (!cidFromPath) {
+			console.log('No IPFS subpath detected')
+			return
+		}
+
+		try {
+			const v1CidString = CID.parse(cidFromPath).toV1().toString()
+			return `${location.protocol}//${v1CidString}.ipfs.${location.host}`
+		} catch (error) {
+			console.log('Failed to transform string to CID V1', `${error}`)
+			return
+		}
```

Comment:
We shouldn't be using `try/catch` for code flow, and if the path segment following `/ipfs/` isn't a CID then we shouldn't try to redirect.
```suggestion
		const cid = CID.asCID(cidFromPath)
		if (!cid) return
		const v1CidString = cid.toV1().toString()
		return `${location.protocol}//${v1CidString}.ipfs.${location.host}`
```
Double check this code works as the docs say and returns `undefined` in the case where the path segment is not a CID, and returns a correct URL when it is either a CIDv0 or CIDv1, but this lets us avoid using a try/catch.

Reply:
the `asCID` function does not work with CID as strings, only with CID objects generated by `CID.parse` or similar libraries

Reply:
This makes me sad, but we can move back to the regex then.  We still shouldn't be using `try/catch` for control flow, so before we call `CID.parse(...)` we should already be very confident we have an IPFS content hash in `cidFromPath`, which means using that regex you had previously.